pkg/ingester/errors.go

```diff
@@ -28,6 +28,20 @@ var (
 		"the ingester is currently too busy to process queries, try again later")
 )
 
+type safeToWrap interface {
+	safeToWrap()
+}
+
+type safeToWrapError struct {
+	err error
+}
+
+func (s safeToWrapError) safeToWrap() {}
+
+func (s safeToWrapError) Error() string {
+	return s.err.Error()
+}
```

Comment:
Can we just embed `error` in the type and avoid having to implement this method?